src/common/TelemetryData.ts

```diff
@@ -0,0 +1,36 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+export class TelemetryData {
```

Comment:
When using TypeScript, prefer using an interface instead of requiring a class. This will give you type safety without generating extra js code.
```ts
export interface ITelemetryData {
   eventName: string;
   properties?: Record<string, string>;
   measurements?: Record<string, number>;
}
```

Then, when constructing, just use regular json notation:
```ts
let data: ITelemetryData = {
   eventName: 'myEvent',
   properties: {
      someKind: 'that-kind',
   }
};
```

But so far (haven't finished this PR), I think this is likely not even needed at all if you just simplify your code to make the telemetry arguments inline.